specification/under_development/profiles/Base_Profile.md

```diff
@@ -0,0 +1,484 @@
+depending on the targeted backend. The profile specification defines how to
+leverage and combine the available instructions to express a program, but does
+not dictate which quantum instructions may be used. Targeting a program to a
+specific backend requires choosing a suitable profile and quantum instruction
+set (QIS). Both can be chosen largely independently, though certain instruction
+sets may be incompatible with this (or other) profile(s). The section on the
+[quantum instruction set](#quantum-instruction-set) below defines the criterion
+for a QIS to be compatible with the Base Profile. More information about the
+role of the QIS, recommendations for front- and backend providers, as well as
+the distinction between runtime functions and quantum instructions can be found
+in [this document](../Instruction_Set.md).
+
+**Bullet 2: Measurements** <br/>
+
+The second requirement should be taken to mean that a Base Profile compliant
+program does *not* apply instructions to a qubit after it has been measured; all
+instructions to transform the quantum state can be applied before performing any
+measurements. It specifically also implies the following:
+
+- There is no need for the quantum processor ([QPU](../Execution.md)) to be able
+to measure only a subset of all available qubits at a time.
+- Executing a Base Profile compliant program does not require support for
+applying quantum instructions dependent on measurement outcomes.
+
+**Bullet 3: Program output** <br />
+
+The specification of QIR and all its profiles needs to permit to accurately
+reflect the program intent. This includes being able to define and customize the
+program output. The Base Profile specification hence requires explicitly
+expressing which values/measurements are returned by the program and in which
+order. How to express this is defined in the section on [output
+recording](#output-recording).
+
+While it is sufficient for the QPU to do a final measurement of all qubits in a
+predefined order at the end of the program, only the selected subset will be
+reflected in the produced output schema. A suitable output schema can be
+generated in a post-processing step after the computation on the quantum
+processor itself has completed; customization of the program output hence does
+not require support on the QPU itself.
+
+The defined [output schemas](../output_schemas/) provide different options for
+how a backend may express the computed value(s). The exact schema can be freely
+chosen by the backend and is identified by a string label in the produced
+schema. Each output schema contains sufficient information to allow quantum
+programming frameworks to generate a user friendly presentation of the returned
+values in the requested order, such as, e.g., a histogram of all results when
+running the program multiple times.
+
+## Program Structure
+
+A Base Profile compliant program is defined in the form of a single LLVM bitcode
```

Comment:
"single LLVM bitcode file" => Bitcode is only one of the possible formats for IR. I guess the human-readable version would be valid too for compliance-checking. "single LLVM file"? "single LLVM IR file"? "single LLVM/QIR file"?

Reply:
No, the human readable format is there for human convenience. However, only the bitcode gives certain guarantees with respect to compatibility. Hence, the requirement is actually explicitly that it is a bitcode file rather than a .ll file. At the same time, it would be rather hard to write the spec without showing llvm snippets in human readable form. I was hoping the paragraph in this section would clarify that the human readable form is merely there for clarity and only the bitcode is expected to be processed.